### PR TITLE
Versioning with new design

### PR DESCRIPTION
**Pull request by ad-47**
Moved PR: https://github.com/VertaAI/modeldb/pull/76 here

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/entities/dataset/PathDatasetComponentBlobEntity.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    this.last_modified_at_source = pathDatasetComponentBlob.getLastModifiedAtSource();
+    this.sha256 = pathDatasetComponentBlob.getSha256();
+    this.md5 = pathDatasetComponentBlob.getMd5();
+  }
+
+  @Id
+  @Column(name = "blob_hash", nullable = false, columnDefinition = "varchar", length = 32)
+  private String blob_hash;
+
+  @Column(name = "path", columnDefinition = "text")
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/entities/dataset/PathDatasetComponentBlobEntity.java]: Doesn't `text` provide some limitations vs varchar? We can potentially place a limit here and use varchar if this is correct.

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  public LabelsMappingEntity(Long repoId, String entityHash, Integer entityType, String label) {
+    this.repo_id = repoId;
+    this.label = label;
+    this.entity_type = entityType;
+    this.entity_hash = entityHash;
+  }
+
+  @Id
+  @Column(name = "repo_id", updatable = false, nullable = false)
+  private Long repo_id;
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java]: There is nothing about repositories on labels. The metadata service is completely independent from the versioning service. You will not see any reference to repository in the API at https://github.com/VertaAI/modeldb/blob/master/protos/protos/public/modeldb/metadata/MetadataService.proto.

File: backend/src/main/java/ai/verta/modeldb/entities/versioning/TagsEntity.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  public TagsEntity(Long repositoryId, String commitHash, String tag) {
+    this.repository_id = repositoryId;
+    this.commit_hash = commitHash;
+    this.tag = tag;
+  }
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "id", updatable = false, nullable = false)
+  private Long id;
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/entities/versioning/TagsEntity.java]: Why is this necessary?

File: backend/src/main/java/ai/verta/modeldb/entities/dataset/PathDatasetComponentBlobEntity.java (hunk 1 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/entities/dataset/PathDatasetComponentBlobEntity.java]: I think text was added to allow longer names, text columns can not be indexed, but I doubt this field will need index. If we do want a limit then varchar with limit will be cleaner.
@conradoverta  limit to 255 chars?

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/entities/dataset/PathDatasetComponentBlobEntity.java]: Is it possible for us one day to support "does any dataset use the S3 bucket B?"? I'm not sure what query would create this, but trying to keep such cases in mind.

File: backend/src/main/java/ai/verta/modeldb/entities/versioning/TagsEntity.java (hunk 3 above)
Comment (ad-47) [reply backend/src/main/java/ai/verta/modeldb/entities/versioning/TagsEntity.java]: Removed

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java (hunk 2 above)
Comment (ad-47) [reply backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java]: Removed

**Comment by ravishetye** [comment]
Please build, deploy the branch, look at kibana and ensure you can do some basic activities through webapp,  on a dev env   before merging

**Comment by ad-47** [comment]
Tested.
